ts/sagas/wallet.ts

```diff
@@ -2,49 +2,97 @@
  * A saga that manages the Wallet.
  */
 
-import { call, Effect, put, takeLatest } from "redux-saga/effects";
+import { call, Effect, fork, put, select, take } from "redux-saga/effects";
 
+import { Option } from "fp-ts/lib/Option";
+import { addCard, deleteCard } from "../api/pagopa";
 import { WalletAPI } from "../api/wallet/wallet-api";
 import {
+  ADD_CARD_REQUEST,
+  CARD_ERROR,
+  DELETE_CARD_REQUEST,
   FETCH_CARDS_REQUEST,
   FETCH_TRANSACTIONS_REQUEST
 } from "../store/actions/constants";
 import { cardsFetched } from "../store/actions/wallet/cards";
 import { transactionsFetched } from "../store/actions/wallet/transactions";
-import { CreditCard } from "../types/CreditCard";
+import { getNewCardData } from "../store/reducers/wallet/cards";
+import { CreditCard, CreditCardId } from "../types/CreditCard";
 import { WalletTransaction } from "../types/wallet";
 
-function* fetchTransactions(
-  loadTransactions: () => Promise<ReadonlyArray<WalletTransaction>>
-): Iterator<Effect> {
+function* fetchTransactions(_: string): Iterator<Effect> {
   const transactions: ReadonlyArray<WalletTransaction> = yield call(
-    loadTransactions
+    WalletAPI.getTransactions
   );
   yield put(transactionsFetched(transactions));
 }
 
-function* fetchCreditCards(
-  loadCards: () => Promise<ReadonlyArray<CreditCard>>
-): Iterator<Effect> {
-  const cards: ReadonlyArray<CreditCard> = yield call(loadCards);
+function* fetchCreditCards(_: string): Iterator<Effect> {
+  const cards: ReadonlyArray<CreditCard> = yield call(WalletAPI.getCreditCards);
   yield put(cardsFetched(cards));
 }
 
+function* addCardSaga(token: string): Iterator<Effect> {
+  const card: Option<CreditCard> = yield select(getNewCardData);
+  if (card.isNone()) {
+    yield put({
+      type: CARD_ERROR,
+      error: Error("Can't find newly entered card")
+    });
+  } else {
+    // call pagopa api to add card
+    const isSuccessful: boolean = yield call(addCard, token, card.value);
+    if (!isSuccessful) {
+      yield put({
+        type: CARD_ERROR,
+        payload: Error("Failed to add card")
+      });
+    }
+  }
+}
+
+function* deleteCardSaga(
+  token: string,
+  cardId: CreditCardId
+): Iterator<Effect> {
+  const isSuccessful: boolean = yield call(deleteCard, token, cardId);
+  if (!isSuccessful) {
+    yield put({
+      type: CARD_ERROR,
+      payload: Error("Failed to delete card!")
+    });
+  }
+}
+
+// this function listens for various wallet-related
+// requests, and handles them as required
+function* watcher(): Iterator<Effect> {
```

Comment:
give it a [more specific name](https://github.com/teamdigitale/italia-app/blob/master/ts/sagas/messages.ts#L195)